### PR TITLE
Add maven archetype for integration tests

### LINKED ISSUE #408: Create a maven archetype to initialize test features
Add a maven archetype to initialize test features

### PR DESCRIPTION
**Pull request by f2par0**
Fixes #408

### CODE REVIEW

File: tooling/camel-karaf-test-feature-archetype/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+### Usage
+
+This tool is using maven archetype:generate plugin, for instance, to 
+initialize a test for camel-Atom , go to camel-karaf/tests/features/ , run
+    
+```shell
+mvn archetype:generate -DfeatureName=Atom -DarchetypeGroupId=org.apache.camel.karaf -DarchetypeArtifactId=camel-karaf-test-feature-archetype -DarchetypeVersion=4.6.0-SNAPSHOT 
+```
+
+Then rename the directory camel-atom-test to camel-atom and adapt pom.xml, fill the supplier class and integration test.
```
Comment (f2par0) [inline tooling/camel-karaf-test-feature-archetype/README.md:17]: this is necessary unless we rename all test directories to match their artifact id (or the other way around)

File: tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                        .log("received message ${body}")
+                        .setBody(constant("OK"));
+    }
+
+    @Override
+    protected void configureProducer(RouteBuilder builder, RouteDefinition producerRoute) {
+
+    }
+
+
```
Comment (essobedo) [inline tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java:55]:
```suggestion
```

File: tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        immediate = true,
+        service = Camel${featureName}RouteSupplier.class
+)
+public class Camel${featureName}RouteSupplier extends AbstractCamelSingleFeatureResultMockBasedRouteSupplier {
+
+
+    @Override
+    public void configure(CamelContext context) {
+
+    }
```
Comment (essobedo) [inline tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java:39]: Should be commented by default as it is not always needed

File: tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import static org.apache.camel.builder.Builder.constant;
+
+import java.util.function.Function;
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.RouteDefinition;
+import org.apache.karaf.camel.itests.AbstractCamelSingleFeatureResultMockBasedRouteSupplier;
+import org.osgi.service.component.annotations.Component;
+
+
```
Comment (essobedo) [inline tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java:27]:
```suggestion
```

File: tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+@Component(
+        name = "karaf-camel-${featureNameLower}-test",
+        immediate = true,
+        service = Camel${featureName}RouteSupplier.class
+)
+public class Camel${featureName}RouteSupplier extends AbstractCamelSingleFeatureResultMockBasedRouteSupplier {
+
+
```
Comment (essobedo) [inline tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java:35]:
```suggestion
```

File: tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public void configure(CamelContext context) {
+
+    }
+
+
```
Comment (essobedo) [inline tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/src/main/java/test/Camel__featureName__RouteSupplier.java:41]:
```suggestion
```

File: tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/pom.xml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    <parent>
+        <groupId>org.apache.camel.karaf</groupId>
+        <artifactId>camel-karaf-features-test</artifactId>
+        <version>${version}</version>
+    </parent>
+
+    <artifactId>camel-${featureNameLower}-test</artifactId>
+    <name>Apache Camel :: Karaf :: Tests :: Features :: ${featureName}</name>
+
+
```
Comment (essobedo) [inline tooling/camel-karaf-test-feature-archetype/src/main/resources/archetype-resources/pom.xml:33]:
```suggestion
```

File: tooling/camel-karaf-test-feature-archetype/README.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# Apache Camel Karaf Archetype
+
+### Introduction
+
+This archetype is to initialize a maven project to test a camel-feature inside karaf using the 
+org.apache.camel.karaf:camel-integration-test
```
[the remark this answers is not in the document]
Comment (f2par0) [reply tooling/camel-karaf-test-feature-archetype/README.md]: it's the name of the maven artifact containing the code of the test framework, I rephrased it

File: tooling/camel-karaf-test-feature-archetype/README.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# Apache Camel Karaf Archetype
```
Comment (essobedo) [inline tooling/camel-karaf-test-feature-archetype/README.md:1]:
```suggestion
# Apache Camel Karaf Integration Test Archetype
```

File: tooling/camel-karaf-test-feature-archetype/pom.xml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.camel.karaf</groupId>
+        <artifactId>tooling</artifactId>
+        <version>4.6.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>camel-karaf-test-feature-archetype</artifactId>
```
Comment (essobedo) [inline tooling/camel-karaf-test-feature-archetype/pom.xml:29]: Please also add a name

**Review by essobedo — APPROVED** [review]
LGTM

**Pull request merged**